src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -802,59 +799,25 @@ InferenceProfiler::Summarize(
   return cb::Error::Success;
 }
 
-void
-InferenceProfiler::MeasurementTimestamp(
-    const TimestampVector& timestamps,
-    std::pair<uint64_t, uint64_t>* valid_range, long long measurement_window_ms)
-{
-  // finding the start time of the first request
-  // and the end time of the last request in the timestamp queue
-  uint64_t first_request_start_ns = 0;
-  uint64_t last_request_end_ns = 0;
-  for (auto& timestamp : timestamps) {
-    uint64_t request_start_time = TIMESPEC_TO_NANOS(std::get<0>(timestamp));
-    uint64_t request_end_time = TIMESPEC_TO_NANOS(std::get<1>(timestamp));
-    if ((first_request_start_ns > request_start_time) ||
-        (first_request_start_ns == 0)) {
-      first_request_start_ns = request_start_time;
-    }
-    if ((last_request_end_ns < request_end_time) ||
-        (last_request_end_ns == 0)) {
-      last_request_end_ns = request_end_time;
-    }
-  }
-
-  // Define the measurement window [client_start_ns, client_end_ns) to be
-  // in the middle of the queue
-  uint64_t measurement_window_ns = measurement_window_ms * 1000 * 1000;
-  uint64_t offset = first_request_start_ns + measurement_window_ns;
-  offset =
-      (offset > last_request_end_ns) ? 0 : (last_request_end_ns - offset) / 2;
-
-  uint64_t start_ns = first_request_start_ns + offset;
-  uint64_t end_ns = start_ns + measurement_window_ns;
-
-  *valid_range = std::make_pair(start_ns, end_ns);
-}
-
 void
 InferenceProfiler::ValidLatencyMeasurement(
-    const TimestampVector& timestamps,
     const std::pair<uint64_t, uint64_t>& valid_range,
     size_t& valid_sequence_count, size_t& delayed_request_count,
     std::vector<uint64_t>* valid_latencies)
 {
   valid_latencies->clear();
   valid_sequence_count = 0;
-  for (auto& timestamp : timestamps) {
+  std::vector<size_t> erase_indices{};
+  for (size_t i = 0; i < all_timestamps_.size(); i++) {
+    const auto& timestamp = all_timestamps_[i];
     uint64_t request_start_ns = TIMESPEC_TO_NANOS(std::get<0>(timestamp));
     uint64_t request_end_ns = TIMESPEC_TO_NANOS(std::get<1>(timestamp));
 
     if (request_start_ns <= request_end_ns) {
       // Only counting requests that end within the time interval
-      if ((request_end_ns >= valid_range.first) &&
-          (request_end_ns <= valid_range.second)) {
+      if (request_end_ns <= valid_range.second) {
```

Comment:
This is slightly big change. Before, we would only include a request if it ended between the window start and window stop. But there are rare cases where a request can end between the time that one window ends and the next window starts. This change now says that if a request ends in that gap, it is considered part of the next window. Basically pretending that there are no gaps between windows, and that the next window is extended to be immediately after the first one, (which in a real execution, there will always be some amount of time between calls to the clock functions and other extraneous logic).